Services/Concurrency/Timer.cs

```diff
@@ -73,5 +76,15 @@ public void Stop()
                 this.log.Info("The timer was already disposed.", () => new { e });
             }
         }
+
+        public void PauseTimer()
+        {
+            this.timer?.Change(Timeout.Infinite, Timeout.Infinite);
+        }
+
+        public void UnPauseTimer()
+        {
+            this.StartIn(this.delay);
```

Comment:
pls init the var in the ctor

Reply:
removed the pause/unpause from the timer, I'll leave these changes to you as discussed.